DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,596 @@
+		return
+	if(target)
+		to_chat(user, "<I> Translocation matrix assigned: Unimatrix 325, grid 006. [get_area(target)]</I>")
+
+/obj/structure/borg_teleporter/Bumped(atom/movable/AM)
+	if(!target || !GLOB.borg_collective.teleporters_allowed)
+		return ..()
+	if(isliving(AM))
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(AM), AM.dir)
+		do_sparks(1, 8, src)
+		visible_message("[AM] enters a transwarp conduit!")
+		AM.forceMove(target)
+		playsound(AM.loc, teleport_sound, 100,1)
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(AM), AM.dir)
+
+/*
+/obj/structure/borg_teleporter/attack_hand(mob/user)
+	. = ..()
+	icon_state = "teleporter-off"
+	target = null
+	var/list/A
+//	A = input(user, "Set Translocation Target", "Translocation shift matrix", A) as null|anything in GLOB.teleportlocs
+	A = borg_collective.poll_drones_for_area()
+//	var/area/thearea = GLOB.teleportlocs[A]
+	var/area/thearea = pick(A)
+	if(!A.len || ISNULL(thearea)
+		thearea = pick(GLOB.teleportlocs)
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(!is_blocked_turf(T))
+			L += T
+	target = pick(L)
+	if(!target)
+		to_chat(user, "Unable to set translocation target, could not locate a suitable subspace exit point.")
+		return
+	to_chat(user, "Coordinates locked: [target.x],[target.y]. Unimatrix 325, grid 006")
+	icon_state = "teleporter-on"
+*/
+/obj/structure/borg_teleporter/proc/activate()
+	icon_state = "teleporter-off"
+	target = null
+	var/list/A = list()
+	A = GLOB.borg_collective.poll_drones_for_teleport()
+	var/area/thearea
+	if(A.len)
+		thearea = pick(A)
+	if(!A.len || !thearea || QDELETED(thearea))
+		var/xd = pick(GLOB.teleportlocs)
+		thearea = GLOB.teleportlocs[xd]
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(!is_blocked_turf(T) && !istype(T, /turf/open/openspace))
+			L += T
+	target = pick(L)
+	if(!target) //Aaaand it still couldn't find an exit turf....somehow...
+		visible_message("Unable to set translocation target, could not locate a suitable subspace exit point. Contact an admin immediately.")
+		return
+	GLOB.borg_collective.message_collective("The collective", "Translocation matrix coordinates locked: [target.x],[target.y] in [get_area(target)]. Unimatrix 325, grid 006. Step through the teleporter to beam down.")
+	icon_state = "teleporter-on"
+
+//Borg tool
+
+/obj/item/borg_tool
+	name = "Primary prosthetic interaction node"
+	desc = "This massive prosthetic houses a vicious claw and a full replicator suite for the rapid conversion of an area into God knows what. Click it while holding it to change its mode."
+	icon = 'DS13/icons/weapons/melee.dmi'
+	item_state = "borgtool"
+	icon_state = "borgtool"
+	lefthand_file = 'DS13/icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'DS13/icons/mob/inhands/weapons/guns_righthand.dmi'
+	resistance_flags = UNACIDABLE
+	var/mode = "assimilate"
```

Comment:
use DEFINES for the modes and make it a number,less memory usage

Reply:
This would fuck up the radial and add bloat to it.